src/client/core/utils/content-editable.js

```diff
@@ -101,8 +105,8 @@ export function getFirstVisibleTextNode (el) {
 
         if (domUtils.isTextNode(curNode) && !isInvisibleTextNode(curNode))
             return curNode;
-        else if (domUtils.isRenderedNode(curNode) &&
-                 (domUtils.isElementNode(curNode) || curNode.childNodes && curNode.childNodes.length)) {
+        else if (domUtils.isRenderedNode(curNode) && hasChildren(curNode) &&
+                 !(domUtils.isElementNode(curNode) && !domUtils.isContentEditableElement(curNode))) {
```

Comment:
can we move some conditions into variables? Not it's hard to understand what happens here